test/fields/collections/Lexical/e2e/main/e2e.spec.ts

```diff
@@ -685,8 +685,12 @@ describe('lexicalMain', () => {
     })
   })
 
-  test('creating a link, then clicking in the link drawer, then saving the link, should preserve cursor position and not move cursor to beginning of richtext field', async () => {
+  /**
+   * A fix was implemented to resolve a limitation in Firefox where the escape key resets the active element to the beginning of the page
+   */
```

Comment:
This describes Firefox's behavior, but what exactly is being done to prevent it?

Reply:
I've updated the test comment, please let me know if this isn't what you were after.

Reply:
I can still reproduce the bug in Firefox. When I press escape, the focus goes to the top of the page.
It's not being captured in CI because we only test chromium.

Reply:
I've also now added tests for both previous element and next element.

Reply:
hmm, I don't know if it's related but this user comments that there is a different behavior in macOS since Catalina OS: https://www.codegram.com/blog/firefox-lets-focus-please/

Reply:
Appreciate it! I'll take a look on my MacBook later today.

Reply:
I was unable to replicate any issues in Firefox on macOS.

Firefox version: 132.0.2 (aarch64)
OS: macOS Sequoia 15.1